### PR TITLE
ls update

### PR DESCRIPTION
**Pull request by Whammo**
ls will accept drive#, wildcards or not.

 Spun off rdir will read directory loaded to arbitrary address.

bs added to display directory without reloading.

 Should more be added to ls?
If so, should it be a variable that you can set if you want it or not?
Would that be overkill?

### CODE REVIEW

**Comment by jkotlinski** [comment]
I would maybe (?) drop bs as it does not seem essential, the other stuff seems great.
It would be good to have usage of `ls` documented. Probably it would go well in 4.17 Disk I/O subsection.

**Comment by jkotlinski** [comment]
Another thing: you could add a note about the change to CHANGELOG.md!
Thanks in advance 😊

**Comment by Whammo** [comment]
That seems to be good- I think I've sorted out the docs

**Comment by Whammo** [comment]
I think I'm done for now.
 Congrats to those of you who do this sort of thing for a living. I need a better workflow.  :)

**Comment by Whammo** [comment]
> Better prg without dot and texttt without spaces

Good call.

**Comment by jkotlinski** [comment]
Thank you for the contribution! I will try to test it soon.

File: docs/anatomy.tex [hunk 1]
```diff
@@ -40,7 +40,7 @@ \section{Code}
 The code section contain pure 6502 machine code.
 
 \begin{description}
-\item[20 cf 0e ( jsr \$ecf )] \$ecf is the adress of the \texttt{lit} code. \texttt{lit} copies the two following bytes to parameter stack.
+\item[20 cf 0e ( jsr \$ecf )] \$ecf is the address of the \texttt{lit} code. \texttt{lit} copies the two following bytes to parameter stack.
```
Comment (jkotlinski) [inline docs/anatomy.tex:43]: nice catch!

File: docs/words.tex [hunk 2]
```diff
@@ -378,8 +378,10 @@ \section{Disk I/O}
 \item[required ( filenameptr filenamelength -- )] Like included, except that load is skipped if the file is already loaded.
 \item[loadb ( filenameptr filenamelength dst -- endaddr )] Load binary block to dst. Returns 0 on failure, otherwise address after last written byte.
 \item[saveb ( start end filenameptr filenamelength -- )] Save binary block.
-\item[device ( device\# -- )] Switches the active device. 8 to 11 are valid device\#'s.
+\item[device ( device\# -- )] Switches the active device. 8 to 11 are valid device \#'s.
```
Comment (jkotlinski) [inline docs/words.tex:381]: I realize now that a better change would be to delete the sentence "8 to 11 are valid device\#'s."  It is misleading, other device #'s are valid too.

File: forth_src/debug.fs [hunk 3]
```diff
@@ -82,11 +82,14 @@ space 1+ loop drop
 dup $7f and $20 < if drop '.' then
 emit 1+ loop cr loop
 last-dump ! base ! ;
-
 : n last-dump @ dump ;
 
-: more $d6 c@ $18 = if $12 emit
-." more" $92 emit key drop page then ;
+true value more?
```
Comment (jkotlinski) [inline forth_src/debug.fs:87]: I think I'd leave this enabled always? Do you have any scenario in mind when one would like to set it to false?

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+ - LS now accepts wildcards, drive #'s or not.
+ - Made MORE optional via value MORE?
+### Added
+ - RDIR will display directory formatted data anywhere in memory.
+ - constants TRUE and FALSE
+ - value MORE?  default value TRUE
```
[the remark this answers is not in the document]
Comment (Whammo) [reply CHANGELOG.md]: Cool.

File: forth_src/ls.fs [hunk 5]
```diff
@@ -1,7 +1,37 @@
...
+begin ?dup while        \ abort if addr = 0
+more 2+ dup @ . 2+      \ skip line link, print blocks
+begin dup c@ ?dup while \ eol?
+emit 1+ repeat 1+ cr    \ print line, eol
+dup c@ 0= if c@ then    \ if eof then addr = 0
+repeat ;                
+
+: ls parse-name ?dup if  \ accepts wildcards or not 
+else drop s" $"  
+then here loadb drop here rdir ;
```
[the remark this answers is not in the document]
Comment (Whammo) [reply forth_src/ls.fs:13]: Did I do that?  :)

File: forth_src/base.fs [hunk 6]
```diff
@@ -85,6 +85,8 @@ msb lda,x lsb sta,x
 dup code lda,# 100/ ldy,#
 ['] pushya jmp, ;
 : constant value ;
+-1 constant true
```
[the remark this answers is not in the document]
Comment (Whammo) [reply forth_src/base.fs:88]: I suppose they could lead to words like ISTRUE?
 *poof*

File: forth_src/debug.fs (hunk 3 above)
Comment (Whammo) [reply forth_src/debug.fs]: Just a nicety on top of a nicety. I get that  :)

File: docs/words.tex (hunk 2 above)
Comment (Whammo) [reply docs/words.tex]: DEVICE is somewhat less ambiguous.

File: CHANGELOG.md [hunk 7]
```diff
@@ -6,7 +6,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
-###Fixed
+### Changed
+ - LS now accepts wildcards, drive #'s or not.
+ - Made MORE optional via value MORE?
```
Comment (jkotlinski) [inline CHANGELOG.md:11]: This line should now be removed.

Comment (Whammo) [reply CHANGELOG.md]: Done.

**Comment by jkotlinski** [comment]
Great! Thank you for the contribution :)

**Pull request merged**